timelock-server/src/suiteTest/java/com/palantir/atlasdb/timelock/suite/DbTimeLockSingleLeaderPaxosSuite.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.suite;
+
+import static com.palantir.atlasdb.timelock.TemplateVariables.generateThreeNodeTimelockCluster;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.junit.Rule;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Suite;
+
+import com.github.peterwippermann.junit4.parameterizedsuite.ParameterizedSuite;
+import com.google.common.collect.ImmutableSet;
+import com.palantir.atlasdb.timelock.ImmutableTestableTimelockServerConfiguration;
+import com.palantir.atlasdb.timelock.MultiNodePaxosTimeLockServerIntegrationTest;
+import com.palantir.atlasdb.timelock.TemplateVariables;
+import com.palantir.atlasdb.timelock.TestableTimelockCluster;
+import com.palantir.atlasdb.timelock.TestableTimelockServerConfiguration;
+import com.palantir.timelock.config.PaxosInstallConfiguration;
+
+@RunWith(ParameterizedSuite.class)
+@Suite.SuiteClasses(MultiNodePaxosTimeLockServerIntegrationTest.class)
+public class DbTimeLockSingleLeaderPaxosSuite {
+    private static final Iterable<TemplateVariables> TEMPLATE_VARIABLES = generateThreeNodeTimelockCluster(
+            9086,
+            builder -> builder.clientPaxosBuilder(builder.clientPaxosBuilder()
+                    .isUseBatchPaxosTimestamp(false)
+                    .isBatchSingleLeader(true))
+                    .leaderMode(PaxosInstallConfiguration.PaxosLeaderMode.SINGLE_LEADER));
+    private static final List<TestableTimelockServerConfiguration> TESTABLE_CONFIGURATIONS = StreamSupport.stream(
+            TEMPLATE_VARIABLES.spliterator(), false)
+            .map(variables -> ImmutableTestableTimelockServerConfiguration.builder()
+                    .templateVariables(variables)
+                    .needsPostgresDatabase(true)
+                    .build())
+            .collect(Collectors.toList());
```

Comment:
It may be worth extracting out a utility method that takes templateVariables and needsPG as parameters? There is similar code in TestableTimelockCluster.

Reply:
Sure, can do